### PR TITLE
CORE-617: In Core Crypto, cryptoWalletManagerCreate() can return NULL.

### PR DESCRIPTION
**Pull request by EBGToo**
Requires Java change.

### CODE REVIEW

File: crypto/BRCryptoWalletManager.c [hunk 1]
```diff
@@ -269,6 +275,12 @@ cryptoWalletManagerCreate (BRCryptoCWMListener listener,
         }
     }
 
+cleanupOnError:
```
Comment (michael-brd) [inline crypto/BRCryptoWalletManager.c]: Yikes! Goto statements? These can't be avoided? These would be the first in our source.

Comment (EBGToo) [reply crypto/BRCryptoWalletManager.c]: I know; maybe the first in my life:
```
static void
doCleanupOnWalletManagerError (BRCryptoWalletManager *cwmPtr) {
  cryptoWalletMangerGive (*cwmPtr);
  *cwmPtr = NULL;
}
```
and then
```
if (NULL = ....) { doCleanupOnWalletManagerError(&cwm); break; }
```
? 

Or goto?

Comment (michael-brd) [reply crypto/BRCryptoWalletManager.c]: Or just do (in the BTC case statement):
```
BRWalletManager bwm = BRWalletManagerNew();
if (NULL != bwm) {
  // normal business
} else {
  // error handling
}
```
Adds one level of indentation, but removes need for goto?

Comment (EBGToo) [reply crypto/BRCryptoWalletManager.c]: Updated.

**Pull request merged**